iothub/service/tests/DigitalTwinsClientTests.cs

```diff
@@ -0,0 +1,471 @@
+using Moq;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.Devices.Tests
+{
+    [TestClass]
+    [TestCategory("Unit")]
+    public class DigitalTwinsClientTests
+    {
+        private const string HostName = "contoso.azure-devices.net";
+        private static readonly string s_validMockAuthenticationHeaderValue = $"SharedAccessSignature sr={HostName}&sig=thisIsFake&se=000000&skn=registryRead";
+        private static readonly string s_connectionString = $"HostName={HostName};SharedAccessKeyName=iothubowner;SharedAccessKey=dGVzdFN0cmluZzE=";
+
+        private static readonly Uri s_httpUri = new($"https://{HostName}");
+        private static readonly RetryHandler s_retryHandler = new(new IotHubServiceNoRetry());
+        private static IotHubServiceClientOptions s_options = new()
+        {
+            Protocol = IotHubTransportProtocol.Tcp,
+            RetryPolicy = new IotHubServiceNoRetry()
+        };
+
+        [TestMethod]
+        public async Task DigitalTwinsClient_GetAsync()
+        {
+            // arrange
+            string digitalTwinId = "foo";
+            var digitalTwin = new BasicDigitalTwin
+            {
+                Id = digitalTwinId,
+            };
+
+            var mockCredentialProvider = new Mock<IotHubConnectionProperties>();
+            mockCredentialProvider
+                .Setup(getCredential => getCredential.GetAuthorizationHeader())
+                .Returns(s_validMockAuthenticationHeaderValue);
+            var mockHttpRequestFactory = new HttpRequestMessageFactory(s_httpUri, "");
+
+            using var mockHttpResponse = new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = HttpMessageHelper.SerializePayload(digitalTwin),
+            };
+            mockHttpResponse.Headers.Add("ETag", "\"AAAAAAAAAAE=\"");
+            
+            var mockHttpClient = new Mock<HttpClient>();
+            mockHttpClient
+                .Setup(restOp => restOp.SendAsync(It.IsAny<HttpRequestMessage>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(mockHttpResponse);
+
+            var digitalTwinsClient = new DigitalTwinsClient(
+                HostName,
+                mockCredentialProvider.Object,
+                mockHttpClient.Object,
+                mockHttpRequestFactory,
+                s_retryHandler);
+
+            // act
+            DigitalTwinGetResponse<BasicDigitalTwin> result =  await digitalTwinsClient.GetAsync<BasicDigitalTwin>(digitalTwinId).ConfigureAwait(false);
+
+            // assert
+            result.DigitalTwin.Id.Should().Be(digitalTwinId);
+        }
+
+        [TestMethod]
+        public async Task DigitalTwinsClient_GetAsync_EmptyTwinIdThrows()
+        {
+            // arrange
+            string digitalTwinId = null;
+            using var serviceClient = new IotHubServiceClient(s_connectionString);
+            DigitalTwinsClient digitalTwinsClient = serviceClient.DigitalTwins;
```

Comment:
You can remove this and call serviceClient.DigitalTwins.GetAsync directly in line 90